### ISSUE TITLE
多卡训练

### ISSUE THREAD

**Issue by Overcautious** [TaoRuijie/TalkNet_ASD#2]
代码能否多卡训练？单张卡显存太小，想用两张2080Ti同时训练，我将代码里的模型加载部分修改了，但是不起作用

          class talkNet(nn.Module):
              def __init__(self, lr = 0.0001, lrDecay = 0.95, **kwargs):
                  super(talkNet, self).__init__()        
                  #self.model = talkNetModel().cuda()
                  # 修改部分
                  self.model = talkNetModel()
                  self.model = nn.DataParallel(self.model, device_ids=[0,1])
                  self.model = self.model.cuda()
                  self.model = self.model.module
                  
                  self.lossAV = lossAV().cuda()
                  self.lossA = lossA().cuda()
                  self.lossV = lossV().cuda()
                  self.optim = torch.optim.Adam(self.parameters(), lr = lr)
                  self.scheduler = torch.optim.lr_scheduler.StepLR(self.optim, step_size = 1, gamma=lrDecay)

**Comment by TaoRuijie** [comment]
我没有试过，可以应该是可以的不过需要怎么改我不清楚欸没有试过。

你可以直接把batchsize调小，设置成1500 甚至1000 就好了，结果上应该基本没有区别，时间可能稍微长一点，但不会超过一天。



For multiple-GPU training, I did not try since I only have one card. So if the memory for your GPU is not big enough, you can also try to reduce the batch size from 2500 to 1500 (or even 1000). Take might take more time but the performance is very similar.

**Comment by Overcautious** [comment]
> 我没有试过，可以应该是可以的不过需要怎么改我不清楚欸没有试过。
> 
> 你可以直接把batchsize调小，设置成1500 甚至1000 就好了，结果上应该基本没有区别，时间可能稍微长一点，但不会超过一天。
> 
> For multiple-GPU training, I did not try since I only have one card. So if the memory for your GPU is not big enough, you can also try to reduce the batch size from 2500 to 1500 (or even 1000). Take might take more time but the performance is very similar.

好的，谢谢

**Comment by shhuangcoder** [comment]
请问多卡的问题解决了吗，我也加了多卡训练的方法，也是不起作用，目的是想把训练的速度提升上去😯

**Comment by JefferyChen453** [comment]
> 请问多卡的问题解决了吗，我也加了多卡训练的方法，也是不起作用，目的是想把训练的速度提升上去😯

可以看下3d-speaker仓库，有对talknet多卡训练的实现
https://github.com/modelscope/3D-Speaker/tree/main/egs/ava-asd/talknet
